filebeat/prospector/prospector.go

```diff
@@ -28,6 +28,7 @@ type Prospector struct {
 	done             chan struct{}
 	states           *file.States
 	wg               sync.WaitGroup
+	channelWg        sync.WaitGroup // Separate waitgroup for channels as not stopped on completion
```

Comment:
TBH I still not fully understand why this is required. But resource management just became somewhat more complex :( 

What bothers me is adding many async/sync code-blocks(utils) trying to manage something adding more and more complexity.